lib/blockscore/base.rb

```diff
@@ -60,8 +60,16 @@ def persisted?
       !id.nil? && !deleted?
     end
 
+    def deleted?
+      attributes.fetch(:deleted, false)
+    end
+
     private
 
+    def already_deleted
```

Comment:
I wonder if a better name for this method would be something like `assert_not_deleted` or `assert_persisted` (although I think the former is probably better)